File: pyg-manager/pyg-manager-interface/src/main/java/com/pyg/manager/service/BrandService.java
package com.pyg.manager.service;

import com.pyg.pojo.TbBrand;
import com.pyg.utils.PageResult;

import java.util.List;
import java.util.Map;

/**
 * Created by on 2018/9/17.
 */
public interface BrandService {

    /**
     * 需求：查询所有品牌
     */
    public List<TbBrand> findAll();
    /**
     * 需求：分页查询品牌数据
     * 参数：Integer page,Integer rows
     * 返回值：包装类对象
     */
    public PageResult findPage(Integer page,Integer rows);
    /**
     * 需求：品牌数据保存
     */
    public void insert(TbBrand brand);
    /**
     * 需求：根据id查询品牌数据
     */
    public TbBrand findOne(Long id);

    /**
     * 需求：品牌数据修改
     */
    public void update(TbBrand brand);
    /**
     * 需求：品牌数据删除
     */
    public void delete(Long[] ids);

    /**
     * 需求：查询品牌数据，进行下拉框列表展示，进行多项选择
     * [{"id":27,"text":"网络"},{"id":32,"text":"机身内存"}]
     */
    public List<Map> findBrandList();
}

File: pyg-manager-web/src/main/webapp/js/service/brandService.js
//定义服务
//ajax发送请求内置服务
app.service("brandService", function ($http) {
    //定义查询所有
    this.findAll = function () {
        //发送请求
        return $http.get("../brand/findAll");
    };

    //分页查询
    this.findPage = function (page, rows) {
        //使用$http内置服务向后台发送请求，进行品牌列表分页查询
        return $http.get("../brand/findPage/" + page + "/" + rows);
    };

    //添加
    this.add = function (entity) {
        return $http.post("../brand/add", entity)
    };

    //修改
    this.update = function (entity) {
        return $http.post("../brand/update", entity)
    };

    //根据id查询品牌
    this.findOne = function (id) {
        //使用内置服务方式请求，根据id查询品牌数据
        return $http.get("../brand/findOne/" + id);
    };
    //删除
    this.dele = function (ids) {
        //使用内置服务发送请求，实现删除
        return $http.get("../brand/dele/" + ids);
    };
    //查询品牌数据，进行多项选择
    this.findBrandList = function () {
        return $http.get("../brand/findBrandList");
    }

})
